chatsky/llm/prompt.py

```diff
@@ -17,35 +19,98 @@ class PositionConfig(BaseModel):
     Lower number prompts will go before higher number prompts in the
     LLM history.
     """
-
     system_prompt: float = 0
     history: float = 1
     misc_prompt: float = 2
     call_prompt: float = 3
     last_turn: float = 4
 
 
-class Prompt(BaseModel):
+class BasePrompt(BaseModel, ABC):
     """
-    LLM Prompt.
-    Provides position config and allow validating prompts from message or string.
+    Base class for all prompts.
+    Provides common interface for getting Langchain messages.
     """
-
-    message: AnyResponse
     position: Optional[float] = None
+    @abstractmethod
+    async def to_langchain_messages(
+        self, 
+        ctx: Context, 
+        source: str = "human",
+        position_config: Optional[PositionConfig] = None
```

Comment:
These two arguments seem unnecessary.
`position_config` is not used in either of the implementations and `source` doesn't make much sense since all prompts should be `system` by default and implement their own custom `source` rules.